src/main/scala/com/cloudant/clouseau/SupportedAnalyzers.scala

```diff
@@ -122,248 +116,249 @@ object SupportedAnalyzers {
-        case _ =>
+        case None =>
           Some(new NorwegianAnalyzer(IndexService.version))
       }
     case "polish" =>
-      options.get("stopwords") match {
-        case Some(stopwords: List[String]) =>
+      getStopWords(options) match {
+        case Some(stopwords) =>
           Some(new PolishAnalyzer(IndexService.version, stopwords))
-        case _ =>
+        case None =>
           Some(new PolishAnalyzer(IndexService.version))
       }
     case "portuguese" =>
-      options.get("stopwords") match {
-        case Some(stopwords: List[String]) =>
+      getStopWords(options) match {
+        case Some(stopwords) =>
           Some(new PortugueseAnalyzer(IndexService.version, stopwords))
-        case _ =>
+        case None =>
           Some(new PortugueseAnalyzer(IndexService.version))
       }
     case "romanian" =>
-      options.get("stopwords") match {
-        case Some(stopwords: List[String]) =>
+      getStopWords(options) match {
+        case Some(stopwords) =>
           Some(new RomanianAnalyzer(IndexService.version, stopwords))
-        case _ =>
+        case None =>
           Some(new RomanianAnalyzer(IndexService.version))
       }
     case "russian" =>
-      options.get("stopwords") match {
-        case Some(stopwords: List[String]) =>
+      getStopWords(options) match {
+        case Some(stopwords) =>
           Some(new RussianAnalyzer(IndexService.version, stopwords))
-        case _ =>
+        case None =>
           Some(new RussianAnalyzer(IndexService.version))
       }
     case "classic" =>
-      options.get("stopwords") match {
-        case Some(stopwords: List[String]) =>
+      getStopWords(options) match {
+        case Some(stopwords) =>
           Some(new ClassicAnalyzer(IndexService.version, stopwords))
-        case _ =>
+        case None =>
           Some(new ClassicAnalyzer(IndexService.version))
       }
     case "standard" =>
-      options.get("stopwords") match {
-        case Some(stopwords: List[String]) =>
+      getStopWords(options) match {
+        case Some(stopwords) =>
           Some(new StandardAnalyzer(IndexService.version, stopwords))
-        case _ =>
+        case None =>
           Some(new StandardAnalyzer(IndexService.version))
       }
     case "email" =>
-      options.get("stopwords") match {
-        case Some(stopwords: List[String]) =>
+      getStopWords(options) match {
+        case Some(stopwords) =>
           Some(new UAX29URLEmailAnalyzer(IndexService.version, stopwords))
-        case _ =>
+        case None =>
           Some(new UAX29URLEmailAnalyzer(IndexService.version))
       }
     case "perfield" =>
       val fallbackAnalyzer = new StandardAnalyzer(IndexService.version)
-      val defaultAnalyzer: Analyzer = options.get("default") match {
-        case Some(defaultOptions) =>
-          createAnalyzerInt(defaultOptions) match {
+      // The "default" field is always a string
```

Comment:
Unfortunately, this cannot always be assumed safely.  I have tried to run CouchDB's `mango-test` target for testing the changes with some `text` searches, and the [`16-index-selectors-test.IndexSelectorText.test_uses_partial_index_with_non_indexable_selector`](https://github.com/apache/couchdb/blob/eb1a5881bb536453a355f16dcb09b223cfaec41e/src/mango/test/16-index-selectors-test.py#L284) test caused an exception (as well as HTTP 500) at this point:

```console
[actor:13] ERROR overlock.threadpool.ErrorLoggedThread - Exception was thrown in thread actor:13
java.lang.ClassCastException: scala.collection.immutable.$colon$colon cannot be cast to java.lang.String
	at com.cloudant.clouseau.SupportedAnalyzers$$anonfun$1.apply(SupportedAnalyzers.scala:359)
	at com.cloudant.clouseau.SupportedAnalyzers$$anonfun$1.apply(SupportedAnalyzers.scala:359)
	at scala.Option.map(Option.scala:133)
	at com.cloudant.clouseau.SupportedAnalyzers$.createAnalyzerInt(SupportedAnalyzers.scala:359)
	at com.cloudant.clouseau.SupportedAnalyzers$.createAnalyzerInt(SupportedAnalyzers.scala:99)
	at com.cloudant.clouseau.SupportedAnalyzers$.createAnalyzer(SupportedAnalyzers.scala:81)
	at com.cloudant.clouseau.IndexService$.start(IndexService.scala:906)
	at com.cloudant.clouseau.IndexManagerService$$anonfun$handleCall$1$$anonfun$apply$1.apply$mcV$sp(IndexManagerService.scala:120)
	at com.cloudant.clouseau.IndexManagerService$$anonfun$handleCall$1$$anonfun$apply$1.apply(IndexManagerService.scala:120)
	at com.cloudant.clouseau.IndexManagerService$$anonfun$handleCall$1$$anonfun$apply$1.apply(IndexManagerService.scala:120)
	at com.yammer.metrics.scala.Timer.time(Timer.scala:17)
	at com.cloudant.clouseau.IndexManagerService$$anonfun$handleCall$1.apply(IndexManagerService.scala:119)
	at com.cloudant.clouseau.IndexManagerService$$anonfun$handleCall$1.apply(IndexManagerService.scala:118)
	at scalang.FunProcess$$anon$2.run(FunProcess.scala:60)
	at org.jetlang.core.BatchExecutorImpl.execute(BatchExecutorImpl.java:11)
	at org.jetlang.fibers.PoolFiber.flush(PoolFiber.java:82)
	at org.jetlang.fibers.PoolFiber.access$100(PoolFiber.java:21)
	at org.jetlang.fibers.PoolFiber$1.run(PoolFiber.java:38)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
	at overlock.threadpool.ErrorLoggedThread.run(NamedThreadFactory.scala:40)
[actor:13] ERROR clouseau.main - Uncaught exception: scala.collection.immutable.$colon$colon cannot be cast to java.lang.String
```

The reason is that the analyzer passed over there is not a string but a list.  This is obviously a mistake, but note that the current code can somehow get away with it without raising an exception.  For the backwards compatibility, it should be understood how such cases are handled and assess if it is worth to maintain the respective old behavior.

Reply:
Perhaps things like that could be prevented by implementing a validation on the Dreyfus (CouchDB) side and present the user with HTTP 400.

Reply:
I printed out options used to construct analyzer while running mango test suite

I saw only following "default" values
 
- keyword
- List(location)

However you are right even if it does look like a mistake we cannot just introduce strict validation.